<?xml version="1.0"?>
<!-- Launches the ROS-based simulation, which uses the real mpc_ros nodes talking to a PyBullet sim environment. -->

<launch>
  <arg name="config"/>
  <arg name="ctrl_config" default="default" doc="alternative controller config overwriting the one in config"/>
  <arg name="planner_config" default="default" doc="alternative controller config overwriting the one in config"/>
  <arg name="use_mpc" default="false" doc="use mpc as controller"/>
  <arg name="logging_sub_folder"  default="default" doc="save data to a sub folder if set"/>

  <!-- Simulation -->
  <include file="$(find mmseq_run)/launch/isaac_sim.launch">
    <arg name="config" value="$(arg config)" />
  </include>
  
  <!-- Controller -->
  <node pkg="mmseq_run" type="controller_ros.py" name="controller_idk" output="screen" required="true" args="--config $(arg config) --ctrl_config $(arg ctrl_config) --planner_config $(arg planner_config) --logging_sub_folder $(arg logging_sub_folder)" unless="$(arg use_mpc)"/>
  <node pkg="mmseq_run" type="mpc_ros.py" name="controller_mpc" output="screen" required="true" args="--config $(arg config) --ctrl_config $(arg ctrl_config)  --planner_config $(arg planner_config) --logging_sub_folder $(arg logging_sub_folder)" if="$(arg use_mpc)"/>
  
  <!-- Robot Model -->
  <include file="$(find mmseq_run)/launch/thing_description.launch">
    <!-- <arg name="visual_params" value="$(find ur_description)/config/ur10/visual_parameters.yaml" /> -->
  </include>
</launch>